src/engine/enginebuffer.h

```diff
@@ -387,6 +392,7 @@ class EngineBuffer : public EngineObject {
     QAtomicInt m_iEnableSyncQueued;
     QAtomicInt m_iSyncModeQueued;
     ControlValueAtomic<double> m_queuedSeekPosition;
+    ControlValueAtomic<EngineChannel*> m_pSyncPosition;
```

Comment:
You can use here the qt class 
QAtomicPointer
The variable name is missleading. How about. m_pChannelForSync or something?